common/lib/dependabot/shared_helpers.rb

```diff
@@ -193,14 +193,31 @@ def self.run_helper_subprocess(command:, function:, args:, env: nil,
           trace: response["trace"]
         )
       rescue JSON::ParserError
-        raise HelperSubprocessFailed.new(
-          message: stdout || "No output from command",
-          error_class: "JSON::ParserError",
-          error_context: error_context
-        )
+        raise handle_json_parse_error(stdout, stderr, error_context)
       end
     end
 
+    sig do
+      params(stdout: String, stderr: String, error_context: T::Hash[Symbol, T.untyped])
+        .returns(Dependabot::SharedHelpers::HelperSubprocessFailed)
+    end
+    def self.handle_json_parse_error(stdout, stderr, error_context)
+      # If the JSON is invalid, the helper has likely failed
+      # We should raise a more helpful error message
+      message = if !stdout.strip.empty?
+                  stdout
+                elsif !stderr.strip.empty?
+                  stderr
+                else
+                  "No output from command"
+                end
+      HelperSubprocessFailed.new(
```

Comment:
This will still be the `HelperSubprocessFailed.new` failure class, is there a possibility to have a more specific error grouping?

Reply:
This check is part of the global helper space and can be utilized by any package manager, so the exact reason for the error is unclear. We handle errors and provide appropriate messages in the package manager updaters.

For reference, you can find some error conversions for Yarn in this file: `dependabot-core/npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb`